### PR TITLE
Add ctx.OtherModule(Reverse)DependencyVariantExists.

### PR DESCRIPTION
**Pull request by marstj**
This allows guarding calls to AddVariationDependencies and
AddReverseDependency to register dependencies on optional modules.

### CODE REVIEW

File: module_ctx.go [hunk 1]
```diff
@@ -275,6 +275,17 @@ type BaseModuleContext interface {
...
+	// OtherModuleVariantExists returns true if a module with the specified name
+	// and variant exists. The variant must match the given variations. It must
+	// also match all variations of the current module if reverse is true, or all
+	// the non-local variations of the current module if reverse is false.
+	//
+	// In other words, if reverse is false it checks for the module
+	// AddVariationDependencies would add a dependency on with the same arguments.
+	// If reverse is true and variations is nil, it checks for the module
+	// AddReverseDependency would add a dependency on.
+	OtherModuleVariantExists(variations []Variation, name string, reverse bool) bool
```
Comment (colincross) [inline module_ctx.go:288]: I'd prefer not to take the "reverse" argument.  Can you instead add two separate methods that call a helper with reverse true or false under the hood?  Something like OtherModuleDependencyVariantExists and OtherModuleReverseDependencyVariantExists.

Comment (marstj) [reply module_ctx.go:288]: Ok. Should OtherModuleReverseDependencyVariantExists take a variations argument then, seeing as the corresponding AddReverseDependency doesn't do that?

Comment (colincross) [reply module_ctx.go:288]: Leaving out the variation sounds good for OtherModuleReverseDependencyVariantExists

Comment (marstj) [reply module_ctx.go]: Done, PTAL. I've tested the change with my use case in https://r.android.com/q/topic:prebuilt-variants+author:mast.

**Pull request merged**